packages/kbn-scout/src/playwright/global_hooks/synthtrace_ingestion.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { FullConfig } from 'playwright/test';
+import Url from 'url';
+import { Readable } from 'node:stream';
+import type {
+  ApmFields,
+  Fields,
+  InfraDocument,
+  OtelDocument,
+  Serializable,
+  SynthtraceGenerator,
+} from '@kbn/apm-synthtrace-client';
+import { getLogger, createScoutConfig, measurePerformanceAsync, getEsClient } from '../../common';
+import { ScoutTestOptions } from '../types';
+import {
+  getApmSynthtraceEsClient,
+  getInfraSynthtraceEsClient,
+  getOtelSynthtraceEsClient,
+} from '../../common/services/synthtrace';
+
+export type SynthtraceEvents<T extends Fields> = SynthtraceGenerator<T> | Array<Serializable<T>>;
+
+interface SynthtraceIngestionData {
+  apm: Array<SynthtraceEvents<ApmFields>>;
+  infra: Array<SynthtraceEvents<InfraDocument>>;
+  otel: Array<SynthtraceEvents<OtelDocument>>;
+}
+
+export async function ingestSynthtraceDataHook(config: FullConfig, data: SynthtraceIngestionData) {
+  const log = getLogger();
+
+  const { apm, infra, otel } = data;
+  const hasApmData = apm.length > 0;
+  const hasInfraData = infra.length > 0;
+  const hasOtelData = otel.length > 0;
+  const hasAnyData = hasApmData || hasInfraData || hasOtelData;
+
+  if (!hasAnyData) {
+    log.debug('[setup] no synthtrace data to ingest');
+    return;
+  }
+
+  return measurePerformanceAsync(log, '[setup]: ingestSynthtraceDataHook', async () => {
+    // TODO: This should be configurable local vs cloud
```

Comment:
Q: Do we need the comment? How can we get this setting? I saw it's needed to create the scout config and we use `'local'` in other places

Reply:
@jennypavlova I will take care of implementing it and updating fixtures/helpers in https://github.com/elastic/kibana/issues/203588

Reply:
Thank you @dmlemeshko!